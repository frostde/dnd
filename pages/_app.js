
import App from 'next/app'
import React from 'react'
import withRedux from "next-redux-wrapper";
import { Provider } from "react-redux";

import { makeStore } from '../store/store'

class MyApp extends App {
    render() {
        const { Component, pageProps, store } = this.props
        return (
            <Provider store={store}>
                <Component {...pageProps} />
            </Provider>
        )
    }
}

export default withRedux(makeStore)(MyApp)