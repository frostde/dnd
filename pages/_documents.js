import Document, { Html, Head, Main, NextScript } from 'next/document';
import { ServerStyleSheet, createGlobalStyle } from 'styled-components'

const GlobalStyle = createGlobalStyle`
  @font-face {
    font-family: 'Poppins';
    src: url('Poppins-Black.ttf');
    src: url('Poppins-BlackItalic.ttf');
    src: url('Poppins-Bold.ttf');
    src: url('Poppins-BoldItalic.ttf');
    src: url('Poppins-ExtraBold.ttf');
    src: url('Poppins-ExtraBoldItalic.ttf');
    src: url('Poppins-ExtraLight.ttf');
    src: url('Poppins-ExtraLightItalic.ttf');
    src: url('Poppins-Italic.ttf');
    src: url('Poppins-Light.ttf');
    src: url('Poppins-LightItalic.ttf');
    src: url('Poppins-Medium.ttf');
    src: url('Poppins-MediumItalic.ttf');
    src: url('Poppins-Regular.ttf');
    src: url('Poppins-SemiBold.ttf');
    src: url('Poppins-SemiBoldItalic.ttf');
    src: url('Poppins-Thin.ttf');
    src: url('Poppins-ThinItalic.ttf');
  }
`

export default class MyDocument extends Document {
  static async getInitialProps(ctx) {
    const sheet = new ServerStyleSheet()
    const originalRenderPage = ctx.renderPage

    try {
      ctx.renderPage = () =>
        originalRenderPage({
          enhanceApp: App => props => sheet.collectStyles(<App {...props} />),
        })

      const initialProps = await Document.getInitialProps(ctx)
      return {
        ...initialProps,
        styles: (
          <>
            {initialProps.styles}
            {sheet.getStyleElement()}
          </>
        ),
      }
    } finally {
      sheet.seal()
    }
  }

  render() {
    return (
      <Html>
        <Head>
          <meta charSet="utf-8" />
          <meta
            name="viewport"
            content="width=device-width, initial-scale=1, shrink-to-fit=no, user-scalable=0"
          />
        </Head>
        <body>
          <GlobalStyle />
          <Main />
          <NextScript />
        </body>
      </Html>
    )
  }
}