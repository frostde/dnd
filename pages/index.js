import React from 'react'
import Head from 'next/head'
import { connect } from "react-redux";

import { getClasses } from '../actions/classes'

export const Home = ({ classes, getClasses }) => {

    React.useEffect(() => {
        getClasses()
    }, [])

    return (
        <div>
            <Head>
                <title>Home</title>
                <link rel="icon" href="/favicon.ico" />
            </Head>
            <div>
            {
                classes.map((element, i) => {
                    return (<div class="test" key={i}>{element.name}</div>)
                })
            }
            </div>
        </div>
    )

}

const mapDispatchToProps = (dispatch) => {
    return {
        getClasses: () => {
            dispatch(getClasses())
        }
    }
}

const mapStateToProps = (state, props) => {
    return {
        classes: state.classes.list
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(Home)
