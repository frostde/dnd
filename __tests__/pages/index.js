import React from 'react'
import { Home } from '../../pages/index'
import { shallow } from 'enzyme'

const setup = function () {
    const props = {
        getClasses: jest.fn(),
        classes: [
            {
                index: "barbarian",
                name: "Barbarian",
                url: "/api/classes/barbarian"
            },
            {
                index: "wizard",
                name: "Wizard",
                url: "/api/classes/wizard"
            }
        ],
    }
    const enzymeWrapper = shallow(<Home {...props} />)

    return {
        props,
        enzymeWrapper
    }
}


describe('component', () => {

    it('renders correctly', () => {
        const { enzymeWrapper } = setup()
        expect(enzymeWrapper.debug()).toMatchSnapshot()
    })

    it('calls action to fetch classes', () => {
        const { props } = setup()

        const mockUseEffect = () => {
            useEffect.mockImplementationOnce(f => f());
        };
        let useEffect = jest.spyOn(React, "useEffect");
        mockUseEffect();

        shallow(<Home {...props}/>)
        
        expect(props.getClasses).toHaveBeenCalled()
    })

    // it('should render 2 divs if 2 classes are present', () => {
    //     const { enzymeWrapper } = setup()
    //     expect(enzymeWrapper.find('test').length).toBe(2)
    // })
})
