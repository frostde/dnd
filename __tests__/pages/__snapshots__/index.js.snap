// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component renders correctly 1`] = `
"<div>
  <Head>
    <title>
      Home
    </title>
    <link rel=\\"icon\\" href=\\"/favicon.ico\\" />
  </Head>
  <div>
    <div class=\\"test\\">
      Barbarian
    </div>
    <div class=\\"test\\">
      Wizard
    </div>
  </div>
</div>"
`;
